src/staticanalysis/bot/static_analysis_bot/workflow.py

```diff
@@ -178,22 +185,24 @@ def run(self, revision):
                     analyzers.append(ClangFormat)
                 else:
                     logger.info('Skip clang-format')
+
             if revision.has_infer_files:
                 if INFER in self.analyzers:
                     analyzers.append(Infer)
                     logger.info('Setup Taskcluster infer build...')
                     setup_infer(self.index_service)
                 else:
                     logger.info('Skip infer')
+
             if not (revision.has_clang_files or revision.has_clang_files):
                 logger.info('No clang or java files detected, skipping mach, infer and clang-*')
 
             # Setup python environment
             logger.info('Mach lint setup...')
             cmd = ['gecko-env', './mach', 'lint', '--list']
             out = run_check(cmd, cwd=settings.repo_dir)
-            assert 'error: problem with lint setup' not in out.decode('utf-8'), \
-                'Mach lint setup failed'
+            if 'error: problem with lint setup' in out.decode('utf-8'):
+                raise AnalysisException('mach', 'Mach lint setup failed')
```

Comment:
Maybe use different codes for the different mach commands? This way you can have more detailed statistics.